.github/scripts/convert_to_notebook.py

```diff
@@ -0,0 +1,51 @@
+from sphinx_gallery.py_source_parser import split_code_and_text_blocks
+from sphinx_gallery.notebook import jupyter_notebook, save_notebook
+from sphinx_gallery import gen_gallery
+from pathlib import Path
+import copy
+
+
```

Comment:
I guess one major problem is the rst syntax not appearing nicely in colab, but I appreciate this may take some effort to fix!

<img width="1050" height="280" alt="Image" src="https://github.com/user-attachments/assets/5018a0bc-0b1f-4108-a83c-d48274d71433" />